src/main/java/org/apache/xml/security/encryption/XMLCipherUtil.java

```diff
@@ -271,93 +272,42 @@ public static KeyAgreementParameters constructAgreementParameters(String agreeme
      * Construct a KeyDerivationParameter object from the given keyDerivationMethod and keyBitLength
      *
      * @param keyDerivationMethod element with the key derivation method data
-     * @param keyBitLength  expected derived key length
+     * @param keyBitLength        expected derived key length in bits
      * @return KeyDerivationParameters data
-     * @throws XMLSecurityException if the keyDerivationMethod is not supported or invalid parameters are provided
+     * @throws XMLSecurityException if the invalid key derivation parameters are provide
```

Comment:
I updated the code now to ensure that only the XMLEncryptionException can be thrown. I made changes to the XMLCipherUtil.constructKeyDerivationParameter method only and retained the XMLSecurityException in the interface KeyDerivationMethod.getKDFParams; because the interface implementation may be used for key derivation functions used for other purposes than just the encryption in the future.